diesel/src/pg/expression/functions.rs

```diff
@@ -2279,3 +2279,33 @@ define_sql_function! {
     #[sql_name = "row_to_json"]
     fn row_to_json<R: RecordOrNullableRecord + MaybeNullableValue<Json>>(record: R) -> R::Out;
 }
+
+#[cfg(feature = "postgres_backend")]
+define_sql_function! {
+    /// This form of jsonb_object takes keys and values pairwise from two separate arrays.
+    /// In all other respects it is identical to the one-argument form.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # include!("../../doctest_setup.rs");
+    /// #
+    /// # fn main() {
+    /// #     run_test().unwrap();
+    /// # }
+    /// #
+    /// # fn run_test() -> QueryResult<()> {
+    /// #     use diesel::dsl::{jsonb_path_exists};
+    /// #     use diesel::sql_types::{Array, Nullable, Text,Jsonb, Jsonpath};
+    /// #     use serde_json::Value;
+    /// #     use jsonpath_rust::JsonPath;
+    /// #     let connection = &mut establish_connection();
+    /// let jsonb:Value = serde_json::json!({"a":[1,2,3,4,5]});
+    /// let json_path = jsonpath_rust::path!("$.a[ ? (@ >= 2 && @ <= 4)]");
+    /// let exists = jsonb_path_exists::<Jsonb,Jsonpath,_,_>(jsonb,json_path).get_result::<bool>(connection)?;
```

Comment:
You want to write something like that here:

```suggestion
    /// let exists = diesel::select(jsonb_path_exists::<Jsonb,Jsonpath,_,_>(jsonb,json_path)).get_result::<bool>(connection)?;
```

The underlying issue is that just calling the sql function only creates a query fragment that only represents the function, not a full SELECT statement. By using `diesel::select` you explicitly construct a SELECT statement without from cause, while using whatever you provided as select clause.